### ISSUE TITLE
Add python linter

### ISSUE THREAD

**Issue by jakebeal** [iGEM-Engineering/iGEM-distribution#154]
We need to automate code quality checking. To do this, we should add a GitHub Action that runs [a python linter](https://books.agiliq.com/projects/essential-python-tools/en/latest/linters.html), probably pylint and/or flake8. This will likely throw up a bunch of errors that, in turn, also need resolution and/or known issue marking.

**Comment by GC-repeat** [comment]
Is there any errors/warnings that are commonly ignored ? 
This is the first time I'm doing this so I don't know if I've done it the right way but it runs and throws errors.

**Comment by jakebeal** [comment]
Running and throwing errors is the expected behavior, since there are almost certainly things in the scripts that aren't recommended coding style.  I'd want to try to get the error count to zero.

It does look like it's not getting the root right however (see: https://stackoverflow.com/questions/51095449/pylint-false-positive-e0401-import-errors-in-vscode-while-using-venv) and it's not making it into the subdirectories.

**Comment by GC-repeat** [comment]
I think that I'm good with the subdirectories. Not sure to understand correctly the root thing. What makes it look like it's not getting the root right?

**Comment by jakebeal** [comment]
The pylint errors of this form are showing that its treating `scripts` as a package rather than as the root:
```
scripts/generate_markdown.py:3:0: E0401: Unable to import 'sbol3' (import-error)
```

One other thing --- I'd like to adjust the line-width complaints to allow a more generous line size. PEP8 allows line width up to 99, so let's go for that rather than 79, as screens are much wider than they were in 2001.

**Comment by GC-repeat** [comment]
Ok, I've looked at this error this morning and found that if I add `pip install -r scripts/requirements.txt` for pylint then the error is gone. Any chance that this is the reason or this just a coincidence ?

I'll put the line-width limit to 99.

**Comment by jakebeal** [comment]
Oh, yes! If the dependencies weren't getting installed, that would definitely cause it too!

**Comment by jakebeal** [comment]
Fixed in #204
